schema/default.yaml

```diff
@@ -19,3 +19,12 @@ properties:
     additionalProperties:
       type: string
     description: Provides the default environment variables.
+
```

Comment:
@bradrydzewski We have these open questions regarding the recent default node changes.

Reply:
> 1. How do we know which default child is overridable at stage?

use step, then stage, then globa

> 2. How would trigger work in case repository is override at stage level? Last time we discussed this we decided to keep it only at pipeline level because of issues with trigger.

we can remove repository from the stage level for now

> 3. UI look and feel for the clone and repository settings at stage level?

the UI should stay the same.  we just persist the data to different sections of the yaml

> 4. InputSet's yaml example with default node ?

```
default:
  clone:
    ref: <+input>

stages: ...
```

Branch short syntax

```
default:
  clone:
    ref: refs/heads/main # branch
```

Branch long syntax:

```
default:
  clone:
    ref:
      type: branch
      value: main
```

Tag short syntax:

```
default:
  clone:
    ref: refs/tags/v1.0.0 # tag
```

Tag long syntax:

```
default:
  clone:
    ref:
      type: tag
      value: v1.0.0
```

PR short syntax:

```
default:
  clone:
    ref: refs/pulls/15/head # pr
```

Tag long syntax:

```
default:
  clone:
    ref:
      type: pr
      value: 18
```

> 5. Why repository have ref properties (branch, tag, pr etc) ? Shouldn't they be part of only clone field ?

sorry, forgot to delete them from repository after I moved them to clone.  pushed an update that removes.

> 6. Does repository also have disabled option ? Or that will be controlled by the clone field

only controlled by the clone field

> 7. Don't see strategy (source-branch/merge) and resources option under clone section. We support this currently.

pushed an update to include.